Task3/Task3_Atanas_Kisyov/src/python/main.py

```diff
@@ -0,0 +1,118 @@
+import logging
+
+import argparse
+import psycopg2
+
+
+AVAILABLE_TABLES = (
+    'car_washes',
+    'places',
+    'car_wash_places',
+    'services',
+    'car_wash_services',
+    'users',
+    'accounts',
+    'orders',
+    'order_details',
+)
+INSTRUCTION_MESSAGE = 'This is a list of all available tables in the database. ' \
+                      'Please make an input what you want to see.'
+
+
+def add_arguments():
+    """Initialize available arguments for user to pass to the program"""
+
+    parser = argparse.ArgumentParser(prog='Database Connection')
+    parser.add_argument('Mode', help='r=Read Mode, rw=Read/Write')
+    args = vars(parser.parse_args())
+    return args
+
+
+class Singleton:
+    """Base class to limit pools to one"""
+
+    _instance = None
+
+    def __new__(cls, *args: tuple, **kwargs: dict):
+        """Create an instance only on first call"""
+
+        if not cls._instance:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance
+
+
+class DatabaseConnection:
+    """Database connection interface"""
+
+    DATABASE = 'postgres'
+
+    CONNECTION_SUCCESSFUL_LOGG_MESSAGE = 'Connection successful!'
+    CONNECTION_UNSUCCESSFUL_LOGG_MESSAGE = 'Connection failed! See exception message and try again!'
+
+    def __init__(self):
+        self.__connection = None
+
+    def connect_to_database(self, username: str, password: str):
+        """Establish connection using psycopg2"""
+
+        try:
+            self.__connection = psycopg2.connect(f"dbname={self.DATABASE} \
+            user={username} password={password} host=localhost")
+        except psycopg2.OperationalError as e:
+            logging.critical(self.CONNECTION_UNSUCCESSFUL_LOGG_MESSAGE)
+            logging.critical(e)
+            exit(1)
+
+        logging.info(self.CONNECTION_SUCCESSFUL_LOGG_MESSAGE)
+        return self.__connection
+
+
+class ConnectionPool(Singleton):
+    """Connection pool"""
+
```

Comment:
Also, please check the recording. I provided nice info about module structure (`../car_wash/db/pool.py`). 

I want to see in main:

```
from car_wash.db import pool

.....

def main():

      with pool.ro_pool.connect() as con:
                con.execute(....)